### PR TITLE
Feature/imu

### PR DESCRIPTION
**Pull request by srao32**

### CODE REVIEW

File: Software/DumptruckUltra-MCU/hw/imu/imu.hpp [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+private:
+    TaskHandle_t imuTaskHandle;
+    const std::shared_ptr<Hardware::I2C::I2CBusManager> i2cBus;
+    const std::function<void(AccelerometerData &)> sendAccelData;
+    const std::function<void(GyroscopeData &)> sendGyroData;
+
+    auto dataReadyCallback() -> void;
+    auto imuTask() -> void;
+
+    static constexpr UBaseType_t IMU_TASK_PRIORITY{3};
```
Comment (bennowotny) [inline Software/DumptruckUltra-MCU/hw/imu/imu.hpp]: For consistency, use `tskIDLE_PRIORITY+2`.

File: Software/DumptruckUltra-MCU/hw/proc/proc_setup.hpp [hunk 2]
```diff
@@ -3,13 +3,41 @@
...
+constexpr cyhal_gpio_t I2C_SCL(P9_0);
+constexpr cyhal_gpio_t I2C_SDA(P9_1);
+constexpr cyhal_gpio_t IMU_INT(P9_2);
+constexpr cyhal_gpio_t DIST_INT(P9_3);
+
+constexpr cyhal_gpio_t PressureSensorADC(P10_0);
+constexpr cyhal_gpio_t Servo4_PWM(P10_1);
+constexpr cyhal_gpio_t Servo5_PWM(P10_3);
+constexpr cyhal_gpio_t Servo6_PWM(P10_5);
+constexpr cyhal_gpio_t Servo7_PWM(P12_7);
```
Comment (bennowotny) [inline Software/DumptruckUltra-MCU/hw/proc/proc_setup.hpp]: SUGGESTION: Group by function, not by port.

File: Software/DumptruckUltra-MCU/hw/imu/imu.cpp [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+            }
+#endif
+        }
+    }
+}
+
+auto IMU::dataReadyCallback() -> void {
+    // Wake imuTask when data is ready
+    BaseType_t taskWoken = pdFALSE;
+    vTaskNotifyGiveFromISR(imuTaskHandle, &taskWoken);
```
Comment (bennowotny) [inline Software/DumptruckUltra-MCU/hw/imu/imu.cpp:173]: Add a comment noting why we're using the `Give`/`Take` semantics.

**Review by bennowotny — CHANGES_REQUESTED** [review]

File: Software/DumptruckUltra-MCU/hw/imu/imu.hpp (hunk 1 above)
Comment (srao32) [reply Software/DumptruckUltra-MCU/hw/imu/imu.hpp]: Resolved in [1a61a22](https://github.com/bennowotny/ECE453_DumptruckUltra/pull/14/commits/1a61a22efbc2c0c6be361d32b53a4f34bc8bcca3)

File: Software/DumptruckUltra-MCU/hw/proc/proc_setup.hpp (hunk 2 above)
Comment (srao32) [reply Software/DumptruckUltra-MCU/hw/proc/proc_setup.hpp]: Resolved in [1a61a22](https://github.com/bennowotny/ECE453_DumptruckUltra/pull/14/commits/1a61a22efbc2c0c6be361d32b53a4f34bc8bcca3)

File: Software/DumptruckUltra-MCU/hw/imu/imu.cpp (hunk 3 above)
Comment (srao32) [reply Software/DumptruckUltra-MCU/hw/imu/imu.cpp:173]: Resolved in [1a61a22](https://github.com/bennowotny/ECE453_DumptruckUltra/pull/14/commits/1a61a22efbc2c0c6be361d32b53a4f34bc8bcca3)

**Review by abishekkannan23 — APPROVED** [review]

**Review by bennowotny — APPROVED** [review]

**Pull request merged**